### PR TITLE
frontend-feat: Toll changes are added

### PR DESCRIPTION
**Pull request by spravinkumar9952**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs [hunk 1]
```diff
@@ -74,6 +74,7 @@ import RemoteConfig (RCCarousel(..))
 import Mobility.Prelude (boolToVisibility)
 import Constants 
 import LocalStorage.Cache (getValueFromCache)
+import Debug
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:77]: Remove this import
Check in other files also

File: Frontend/ui-driver/src/Screens/TripDetailsScreen/View.purs [hunk 2]
```diff
@@ -286,12 +286,10 @@ tripDataView push state =
         [ height WRAP_CONTENT
         , width MATCH_PARENT
         , orientation HORIZONTAL
-        , gravity CENTER_VERTICAL
         ][ linearLayout
             [ height WRAP_CONTENT
-            , width WRAP_CONTENT
+            , width $ V $ (EHC.screenWidth unit) / 2
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/TripDetailsScreen/View.purs:291]: Why are we hardcoding view width on many places ?

File: Frontend/ui-driver/src/Screens/TripDetailsScreen/View.purs [hunk 3]
```diff
@@ -358,17 +353,16 @@ tripDataView push state =
...
             ][ textView $
-                [ text (getString TIME_TAKEN)
+                [ text $ getString TOLL_INCLUDED
                 , color Color.black700
-                , margin (MarginBottom 4)
+                , margin $ MarginBottom 4
                 ] <> FontStyle.body5 TypoGraphy
               , textView $
-                [ text state.data.timeTaken
+                [ text $ "₹" <> (show state.data.tollCharge)
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/TripDetailsScreen/View.purs:365]: Better Use currency here
CC @Vignesh-772

File: Frontend/ui-common/src/Components/RateCard/View.purs [hunk 4]
```diff
@@ -318,6 +320,45 @@ fareUpdateView push config =
...
 
+tollOrParkingView :: forall w. (Action -> Effect Unit) -> Config -> PrestoDOM (Effect Unit) w
+tollOrParkingView push config = 
+  linearLayout
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  , orientation VERTICAL
+  , padding $ Padding 20 0 20 160
+  ][
+    textView $[
```
Comment (Rohit4997) [inline Frontend/ui-common/src/Components/RateCard/View.purs:331]: Use common textView here,
These lines can be replaced by 4 lines by using commonTV

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 5]
```diff
@@ -1585,14 +1585,24 @@ estimateHeaderView push state =
...
-        [ text (getString CONFIRM_YOUR_RIDE)
+        [ text$ getString CONFIRM_YOUR_RIDE
         , color Color.black800
         , gravity CENTER_HORIZONTAL
         , height WRAP_CONTENT
         , width MATCH_PARENT
         ] 
         <> FontStyle.h1 TypoGraphy
     , estimatedTimeDistanceView push state
+    , textView $
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1596]: Better use `commonTextView`

File: Frontend/ui-customer/src/Services/Backend.purs [hunk 6]
```diff
@@ -740,13 +740,9 @@ drawMapRoute srcLat srcLng destLat destLng sourceMarkerConfig destMarkerConfig r
...
         callDrawRoute route = do
             case route of
-                Just (Route routes) ->
-                    if (routes.distance <= 50000) then do
-                      void $ liftFlowBT $ runEffectFn9 drawRoute (walkCoordinates routes.points) "LineString" "#323643" true sourceMarkerConfig destMarkerConfig 8 routeType specialLocation
-                      pure route
-                    else do
-                      void $ liftFlowBT $ runEffectFn9 drawRoute (walkCoordinate srcLat srcLng destLat destLng) "DOT" "#323643" false sourceMarkerConfig destMarkerConfig 8 routeType specialLocation
-                      pure route 
+                Just (Route routes) -> do
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Services/Backend.purs:743]: `if (routes.distance <= 50000) then do`
Above check is removed, Where are we handling it now ?

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Services/Backend.purs (hunk 6 above)
Comment (spravinkumar9952) [reply Frontend/ui-customer/src/Services/Backend.purs:743]: its handled in backend

File: Frontend/ui-driver/src/Screens/TripDetailsScreen/View.purs (hunk 2 above)
Comment (spravinkumar9952) [reply Frontend/ui-driver/src/Screens/TripDetailsScreen/View.purs:291]: its not hard coding. we want to show two views like grid. so taken half width

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs (hunk 5 above)
Comment (spravinkumar9952) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1596]: that dont have visibility. otherwise we want to add if else.

**Review by Rohit4997 — DISMISSED** [review]

**Review by witcher-shailesh — DISMISSED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**